### PR TITLE
new upload parameter for overwriting annotations

### PR DESCRIPTION
**Pull request by npelzmann**
# Description

Added a new parameter for the single_upload method to enable overwriting annotations via the API. This has been requested by multiple other users. The option was available via the REST API, but not the python SDK.

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)

## How has this change been tested, please provide a testcase or example of how you tested the change?

Tested upload of image with annotations then deleted part of the annotations manually, and re-uploaded only the annotations.

## Any specific deployment considerations

One of the unittests was already broken on the main branch (`test_download_returns_dataset` in `tests.test_version`. I did not touch this.

## Docs

-   [ ] Docs updated? What were the changes: No changes made to docs.

### CODE REVIEW

File: roboflow/core/project.py
Comment (npelzmann) [reply roboflow/core/project.py]: Done :)

**Review by SolomonLake — APPROVED** [review]
LGTM, after you merge it will be included in the next package release.

File: roboflow/core/project.py
Comment (npelzmann) [reply roboflow/core/project.py]: @SolomonLake I accidentally closed the issue before actually merging. I think you need to approve a second time. Sorry for the inconvenience.

Comment (SolomonLake) [reply roboflow/core/project.py]: Failed to build due to styling -- running `make style` in the root directory should fix the issue.

Comment (npelzmann) [reply roboflow/core/project.py]: Hope the styling checks out now.

Comment (SolomonLake) [reply roboflow/core/project.py]: Looks good, i'll merge the PR now!

**Pull request merged**
